src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs

```diff
@@ -25,25 +25,165 @@ private DesigntimeLicenseContextSerializer()
         /// </summary>
         public static void Serialize(Stream o, string cryptoKey, DesigntimeLicenseContext context)
         {
-            IFormatter formatter = new BinaryFormatter();
+            if (LocalAppContextSwitches.AllowBinaryFormatterUsageInTypeConverterSerialization)
+            {
+                SerializeWithBinaryFormatter(o, cryptoKey, context);
+            }
+            else
+            {
+                using (BinaryWriter writer = new BinaryWriter(o, encoding: Text.Encoding.UTF8, leaveOpen: true))
+                {
+                    writer.Write((sbyte)-1); // flag to identify BinaryWriter
+                    writer.Write(cryptoKey);
+                    writer.Write(context._savedLicenseKeys.Count);
+                    foreach (DictionaryEntry keyAndValue in context._savedLicenseKeys)
+                    {
+                        writer.Write(keyAndValue.Key.ToString());
+                        writer.Write(keyAndValue.Value.ToString());
+                    }
+                }
+            }
+        }
+
+        private static void SerializeWithBinaryFormatter(Stream o, string cryptoKey, DesigntimeLicenseContext context)
```

Comment:
NIT: you can probably just add the `[RequriesUnreferencedCode]` attribute here and that way you can remove the suppresion from ILLink.Suppresions.xml. @eerhardt what is the protocol here? Are we asking new warnings from new PRs in runtime to be annotated or is it fine to suppress and have us do another pass?

Reply:
> Are we asking new warnings from new PRs in runtime to be annotated or is it fine to suppress and have us do another pass?

My general guidance has been:

* if the library is already warning free (or close to it, the case of CoreLib), no new warnings. You need to address them in your change.
* If the library hasn't been worked on yet, and you are just refactoring code, it is fine to update the suppressions file. The suppression was already there and you aren't "adding" new ones. (That's this case)
* If the library hasn't been worked on yet, and you are adding new features - is a little grey. I would assume the new feature should be warning free as it gets written going forward.